MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -89,10 +89,14 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
             return
         }
 
-        switch requestPermissionType {
-        case .location:
-            getPermissionResult(requestPermissionType: requestPermissionType, callbackId: callbackId)
+        if isUserAllowedPermission(customPermissionType: MiniAppCustomPermissionType.deviceLocation) {
```

Comment:
I think we shouldn't change the behavior that we currently have here for the device permission. It makes more sense to me if we leave `requestPermission` to purely handle device permissions so we have a clear separation between custom and device permissions. For example, the host app is able to override the `requestPermission` functionality so it would be more complex if they have to worry about both permission types here, and we won't be able to predict from the JS SDK side how they are handling it.

I think it's better to handle requesting the custom permission/device permission together from the JS SDK side. See the PR from @khairul-alam-licon for the JS SDK: https://github.com/rakutentech/js-miniapp/pull/75/files

Do you think this makes sense?

Reply:
Gotcha 👍 , I didn't have a look into that ticket, sorry 👍